addons/website_slides/models/slide_slide.py

```diff
@@ -315,62 +383,96 @@ def _compute_user_membership_id(self):
-                    record.embed_code = '<iframe src="//drive.google.com/file/d/%s/preview" allowFullScreen="true" frameborder="0"></iframe>' % (record.document_id)
+                    slide.youtube_id = False
             else:
-                record.embed_code = False
+                slide.youtube_id = False
+
+    @api.depends('url', 'document_google_url', 'image_google_url', 'video_url')
+    def _compute_google_drive_id(self):
+        """ Extracts the Google Drive ID from the url based on the slide category. """
 
-    @api.onchange('url')
+        for slide in self:
+            url = slide.url or slide.document_google_url or slide.image_google_url or slide.video_url
+            google_drive_id = False
+            if url:
+                match = re.match(self.GOOGLE_DRIVE_DOCUMENT_ID_REGEX, url)
+                if match and len(match.groups()) == 2:
+                    google_drive_id = match.group(2)
+
+            slide.google_drive_id = google_drive_id
+
+    @api.onchange('url', 'document_google_url', 'image_google_url', 'video_url')
     def _on_change_url(self):
+        """ Keeping a 'onchange' because we want this behavior for the frontend.
+        Changing the document / video external URL will populate some metadata on the form view.
+        The slide metadata are also fetched in create / write overrides to ensure consistency. """
+
         self.ensure_one()
-        if self.url:
-            res = self._parse_document_url(self.url)
-            if res.get('error'):
-                raise UserError(res.get('error'))
-            values = res['values']
-            if not values.get('document_id'):
-                raise UserError(_('Please enter valid Youtube or Google Doc URL'))
-            for key, value in values.items():
-                self[key] = value
-
-    @api.onchange('binary_content')
-    def _on_change_binary_content(self):
-        """ For PDFs, we assume that it takes 5 minutes to read a page.
-            If the selected file is not a PDF, it is an image (You can
-            only upload PDF or Image file) then the slide_category is changed
-            into infographic and the uploaded binary_content is transfered to the
-            image field. (It avoids the infinite loading in PDF viewer)"""
-        if self.binary_content:
-            data = base64.b64decode(self.binary_content)
-            if data.startswith(b'%PDF-'):
-                pdf = PyPDF2.PdfFileReader(io.BytesIO(data), overwriteWarnings=False, strict=False)
-                try:
-                    pdf.getNumPages()
-                except PyPDF2.utils.PdfReadError:
-                    return
-                self.completion_time = (5 * len(pdf.pages)) / 60
-            else:
-                self.slide_category = 'infographic'
-                self.image_1920 = self.binary_content
-                self.binary_content = None
+        if self.url or self.document_google_url or self.image_google_url or self.video_url:
```

Comment:
Isn't it the same ? (url -> other are related)

Reply:
Yep, tried it again with "url" in the view, no changes, nothing triggers when I change the value of my "video_url" field.